contexts/PaymentContext/src/RequestModel/SofortNotificationRequest.php

```diff
@@ -47,4 +48,25 @@ public function setTime( DateTime $time ): self {
 		$this->time = $time;
 		return $this;
 	}
+
+	public static function fromRawRequest( string $content ): ?self {
```

Comment:
The request models are not supposed to contain anything presentation specific and be dumb value objects (no logic). You can argue that since this is a static method, it is not really part of the class of course... Even so I really rather not have this web request parsing code here. It adds web-bound stuff to files that define things that are delivery mechanism agnostic. If you somehow want to enforce the boundaries (or move the Domain bound stuff into its own git repo), this will create problems.